jdaviz/core/astrowidgets_api.py

```diff
@@ -178,8 +182,11 @@ def zoom_level(self):
             raise ValueError('Viewer is still loading, try again later')
 
         if hasattr(self, '_get_real_xy'):
-            i_top = get_top_layer_index(self)
-            image = self.layers[i_top].layer
+            if self.state.reference_data is not None:
+                image, i_ref = get_reference_image_data(self.jdaviz_app, self.reference)

```

Comment:
`i_ref` is not used.
```suggestion
                image, _ = get_reference_image_data(self.jdaviz_app, self.reference)
```